### DISCUSSION TITLE
No development certificates available to code sign app for device deployment.

### DISCUSSION THREAD

**Discussion by SergeiBazunov** [codemagic-ci-cd/codemagic-docs#2305] [Q&A] [answered]
### If you are looking for help, please confirm the following...

- [X] I confirm I have searched the [Docs](https://docs.codemagic.io), [Codemagic Sample Projects](https://github.com/codemagic-ci-cd/codemagic-sample-projects), and GitHub [Discussions](https://github.com/orgs/codemagic-ci-cd/discussions).
- [X] I confirm the software versions on my local computer match the ones in CI environment (e.g. Xcode version, Gradle version). See [here](https://docs.codemagic.io/troubleshooting/common-android-issues/#builds-work-locally-but-fail-on-codemagic)
- [X] I confirm I am able to do a fresh clone of the repository and build the project on my local computer.

### Which mobile framework are you using?

Flutter (Dart)

### Steps to reproduce

I'm currently attempting to build an iOS version of my Flutter project, but I'm encountering an issue during the process
I have tried numerous solutions to this problem but to no avail.

I have linked my Apple Developer account to Codemagic. I have provided Codemagic with the correct App Store Connect API key (.p8 file) along with all necessary identifiers and secrets. In App Store Connect, I have created an App ID that exactly matches the Bundle ID of my application.

I have created and activated the corresponding code signing certificates. For building through Codemagic and subsequent uploading to TestFlight, I use a certificate of type "iOS Distribution".

I have generated the necessary provisioning profiles for use in the build and signing process.

I have tried updating information in Teams > Code signing identities:

Fetch certificate
Generate certificate
A green checkmark appears next to the certificate.
I have tried different code signing methods:

Automatic
Manual (uploaded all necessary certificates and profiles)
I even tried building through codemagic.yaml.

Yet, I encounter the same error every time during the build process.

Could anyone please advise on how to resolve this issue? Thank you very much for your assistance.

### Expected results

ipa artifact

### Actual results

== Building for iOS ==
════════════════════════════════════════════════════════════════════════════════
No valid code signing certificates were found. You can connect to your Apple Developer account by signing in with your Apple ID in Xcode and create an iOS Development Certificate as well as a Provisioning Profile for your project by:
1- Opening the Flutter project's Xcode target with open ios/Runner.xcworkspace.
2- Selecting the 'Runner' project in the navigator, then the 'Runner' target in the project settings.
3- Ensuring a 'Development Team' is selected under Signing & Capabilities > Team. You may need to:
- Log in with your Apple ID in Xcode first.
- Ensure you have a valid, unique Bundle ID.
- Register your device with your Apple Developer Account.
- Let Xcode automatically provision a profile for your app.
4- Building or running your project again.
5- Trusting your newly created Development Certificate on your iOS device via Settings > General > Device Management > [your new certificate] > Trust.

For more information, please visit:
Apple Developer Documentation

Alternatively, run on an iOS simulator without code signing.
════════════════════════════════════════════════════════════════════════════════
No development certificates available to code sign app for device deployment.

### Build id (optional)

65a2de5bed54f1459a8bae51

**Comment by dtrdic** [comment]
hello @nostradamus89, does the bundle ID in your Xcode match the bundle ID you specified in the uploaded provisioning profile? `com.fastest.food.delivery`

**Reply by SergeiBazunov** [reply]
Hello, David! Thank you for reply. I don't use Xcode. I use VSCode in Windows. I sync bundle ID by using App Atore Connect API

**Comment by SergeiBazunov** [comment, marked as answer]
This helps me:
Open your iOS project in an editor and open the project.pbxproj file and set the application PRODUCT_BUNDLE_IDENTIFIER to the bundle id you had created. This is mentioned in multiple places in the project, so make sure to change it everywhere. You can edit the file without a Mac

**Reply by ralon99** [reply]
thank you!!!
omg.. it is such a stupid problem and i've been banging my head against the wall for hours

**Reply by tom11011** [reply]
This worked for me as well, glad I found this post.

**Reply by Amraoui-Chihab** [reply]
Thank you very much . u saved my life !

**Reply by rubody** [reply]
you are a freaking hero i have been trying to find fixing since like 20 hours i didnt even sleep... thank you so much

**Reply by yalpertem** [reply]
i'm also grateful ❤️

**Comment by SergeiBazunov** [comment]
Open your iOS project in an editor and open the project.pbxproj file and set the application PRODUCT_BUNDLE_IDENTIFIER to the bundle id you had created. This is mentioned in multiple places in the project, so make sure to change it everywhere. You can edit the file without a Mac

**Reply by pritamgudekar** [reply]
Hi. Could you please let us know what type of bundle is needed? How it can be mentioned?

**Reply by tom11011** [reply]
The bundle name you told apple, for example-
com.mydomain.mynewapp

**Comment by elarief92** [comment]
still have the same issue

**Reply by dana-prog** [reply]
Is there another tip on how to solve this issue?
i'm getting this error: 
No development certificates available to code sign app for device deployment

I valiadted that bundle id in the xcode project settings matches the bundle id in the profile.
Any other validations i should make?

**Reply by hani-hj1908619** [reply]
@dana-prog  did you end up finding the issue?

**Reply by Equirinya** [reply]
I solved this by also setting the Team and Provisioning Profile Name
